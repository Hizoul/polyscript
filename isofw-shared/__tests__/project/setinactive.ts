import setinactive from "isofw-shared/src/tests/project/setinactive"

setinactive()
