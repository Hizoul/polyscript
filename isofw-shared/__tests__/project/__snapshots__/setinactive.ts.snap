// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`free unused presets after program Update works as planned: Before anything 1`] = `
Object {
  "createState": Object {},
  "currentlyEditing": "",
  "fetching": Object {},
  "getFromServer": [Function],
  "getState": Object {
    "users": Object {
      "u1": Object {
        "_id": "u1",
        "email": "u1",
      },
    },
  },
  "lastFetch": Object {},
  "removeState": Object {},
  "updateState": Object {},
}
`;

exports[`free unused presets after program Update works as planned: after All having been freed up again 1`] = `
Object {
  "createState": Object {
    "freePresetGetter.assistantForm": "c1bbbbbbbbbbbbbbbb398765",
  },
  "currentlyEditing": "b19999999999999999999999",
  "fetching": Object {
    "b19999999999999999999999": false,
  },
  "getFromServer": [Function],
  "getState": Object {
    "presetAssistant": Object {
      "b19999999999999999999999": Object {
        "_id": "b19999999999999999999999",
        "cameras": Array [
          "c1bbbbbbbbbbbbbbbbbbbbbb",
          "c2bbbbbbbbbbbbbbbbbbbbbb",
          "c3bbbbbbbbbbbbbbbbbbbbbb",
        ],
        "isActive": false,
        "name": "pn1",
        "operatorCameraMapping": Array [
          Object {
            "cameras": Array [
              "c1bbbbbbbbbbbbbbbbbbbbbb",
              "c3bbbbbbbbbbbbbbbbbbbbbb",
            ],
            "operator": "u1",
          },
          Object {
            "cameras": Array [
              "c2bbbbbbbbbbbbbbbbbbbbbb",
            ],
            "operator": "u2",
          },
        ],
        "operators": Array [
          "u1",
          "u2",
        ],
        "program": null,
        "shot": 1,
      },
      "undefined": "c1bbbbbbbbbbbbbbbb398765",
    },
    "presets": Object {
      "c1bbbbbbbbbbbbbbbb098765": Object {
        "_id": "c1bbbbbbbbbbbbbbbb098765",
        "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
        "number": 0,
        "project": "-",
      },
      "c1bbbbbbbbbbbbbbbb198765": Object {
        "_id": "c1bbbbbbbbbbbbbbbb198765",
        "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
        "number": 1,
        "project": "-",
      },
      "c1bbbbbbbbbbbbbbbb298765": Object {
        "_id": "c1bbbbbbbbbbbbbbbb298765",
        "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
        "number": 2,
        "project": "-",
      },
      "c1bbbbbbbbbbbbbbbb398765": Object {
        "_id": "c1bbbbbbbbbbbbbbbb398765",
        "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
        "number": 3,
        "project": "-",
      },
    },
    "projects": Object {
      "b19999999999999999999999": Object {
        "_id": "b19999999999999999999999",
        "cameras": Array [
          "c1bbbbbbbbbbbbbbbbbbbbbb",
          "c2bbbbbbbbbbbbbbbbbbbbbb",
          "c3bbbbbbbbbbbbbbbbbbbbbb",
        ],
        "isActive": false,
        "name": "pn1",
        "operatorCameraMapping": Array [
          Object {
            "cameras": Array [
              "c1bbbbbbbbbbbbbbbbbbbbbb",
              "c3bbbbbbbbbbbbbbbbbbbbbb",
            ],
            "operator": "u1",
          },
          Object {
            "cameras": Array [
              "c2bbbbbbbbbbbbbbbbbbbbbb",
            ],
            "operator": "u2",
          },
        ],
        "operators": Array [
          "u1",
          "u2",
        ],
        "program": null,
        "shot": 1,
      },
    },
    "users": Object {
      "u1": Object {
        "_id": "u1",
        "email": "u1",
      },
    },
  },
  "lastFetch": Object {
    "b19999999999999999999999": -2077750800000,
  },
  "removeState": Object {},
  "updateState": Object {
    "projectModelb19999999999999999999999": Object {
      "_id": "b19999999999999999999999",
      "cameras": Array [
        "c1bbbbbbbbbbbbbbbbbbbbbb",
        "c2bbbbbbbbbbbbbbbbbbbbbb",
        "c3bbbbbbbbbbbbbbbbbbbbbb",
      ],
      "isActive": true,
      "name": "pn1",
      "operatorCameraMapping": Array [
        Object {
          "cameras": Array [
            "c1bbbbbbbbbbbbbbbbbbbbbb",
            "c3bbbbbbbbbbbbbbbbbbbbbb",
          ],
          "operator": "u1",
        },
        Object {
          "cameras": Array [
            "c2bbbbbbbbbbbbbbbbbbbbbb",
          ],
          "operator": "u2",
        },
      ],
      "operators": Array [
        "u1",
        "u2",
      ],
      "shot": 1,
    },
    "undefinedb19999999999999999999999": null,
  },
}
`;

exports[`free unused presets after program Update works as planned: after All having been freed up again 2`] = `
Object {
  "currentPage": Object {},
  "dirtyCollections": Object {
    "presetAssistant": true,
    "projects": true,
  },
  "doingQuery": Object {
    "undefinedpresets{\\"camera\\":\\"c1bbbbbbbbbbbbbbbbbbbbbb\\",\\"$limit\\":15,\\"$skip\\":0,\\"$sort\\":{\\"number\\":1,\\"camera\\":1}}": false,
  },
  "lists": Object {
    "list.presetModel": Object {
      "data": Array [
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb098765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 0,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb198765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 1,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb298765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 2,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb398765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 3,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb498765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 4,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb598765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 5,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb698765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 6,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb798765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 7,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb898765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 8,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb998765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 9,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbb1098765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 10,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbb1198765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 11,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbb1298765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 12,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbb1398765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 13,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbb1498765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 14,
          "project": "-",
        },
      ],
      "limit": 15,
      "skip": 0,
      "total": 100,
    },
  },
  "makeQuery": [Function],
  "maxPage": Object {
    "list.presetModel": 7,
  },
  "pageSize": 15,
  "previousQuery": Object {
    "list.presetModel": Object {
      "$limit": 15,
      "$skip": 0,
      "$sort": Object {
        "camera": 1,
        "number": 1,
      },
      "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
    },
  },
}
`;

exports[`free unused presets after program Update works as planned: after reservation of four presets 1`] = `
Object {
  "createState": Object {
    "freePresetGetter.assistantForm": "c1bbbbbbbbbbbbbbbb398765",
  },
  "currentlyEditing": "b19999999999999999999999",
  "fetching": Object {
    "b19999999999999999999999": false,
  },
  "getFromServer": [Function],
  "getState": Object {
    "presetAssistant": Object {
      "undefined": "c1bbbbbbbbbbbbbbbb398765",
    },
    "presets": Object {
      "c1bbbbbbbbbbbbbbbb098765": Object {
        "_id": "c1bbbbbbbbbbbbbbbb098765",
        "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
        "number": 0,
        "project": "-",
      },
      "c1bbbbbbbbbbbbbbbb198765": Object {
        "_id": "c1bbbbbbbbbbbbbbbb198765",
        "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
        "number": 1,
        "project": "-",
      },
      "c1bbbbbbbbbbbbbbbb298765": Object {
        "_id": "c1bbbbbbbbbbbbbbbb298765",
        "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
        "number": 2,
        "project": "-",
      },
      "c1bbbbbbbbbbbbbbbb398765": Object {
        "_id": "c1bbbbbbbbbbbbbbbb398765",
        "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
        "number": 3,
        "project": "-",
      },
    },
    "projects": Object {
      "b19999999999999999999999": Object {
        "_id": "b19999999999999999999999",
        "cameras": Array [
          "c1bbbbbbbbbbbbbbbbbbbbbb",
          "c2bbbbbbbbbbbbbbbbbbbbbb",
          "c3bbbbbbbbbbbbbbbbbbbbbb",
        ],
        "isActive": true,
        "name": "pn1",
        "operatorCameraMapping": Array [
          Object {
            "cameras": Array [
              "c1bbbbbbbbbbbbbbbbbbbbbb",
              "c3bbbbbbbbbbbbbbbbbbbbbb",
            ],
            "operator": "u1",
          },
          Object {
            "cameras": Array [
              "c2bbbbbbbbbbbbbbbbbbbbbb",
            ],
            "operator": "u2",
          },
        ],
        "operators": Array [
          "u1",
          "u2",
        ],
        "shot": 1,
      },
    },
    "users": Object {
      "u1": Object {
        "_id": "u1",
        "email": "u1",
      },
    },
  },
  "lastFetch": Object {
    "b19999999999999999999999": -2077750800000,
    "c1bbbbbbbbbbbbbbbb098765": -2077750800000,
    "c1bbbbbbbbbbbbbbbb198765": -2077750800000,
    "c1bbbbbbbbbbbbbbbb298765": -2077750800000,
    "c1bbbbbbbbbbbbbbbb398765": -2077750800000,
  },
  "removeState": Object {},
  "updateState": Object {
    "projectModelb19999999999999999999999": Object {
      "_id": "b19999999999999999999999",
      "cameras": Array [
        "c1bbbbbbbbbbbbbbbbbbbbbb",
        "c2bbbbbbbbbbbbbbbbbbbbbb",
        "c3bbbbbbbbbbbbbbbbbbbbbb",
      ],
      "isActive": true,
      "name": "pn1",
      "operatorCameraMapping": Array [
        Object {
          "cameras": Array [
            "c1bbbbbbbbbbbbbbbbbbbbbb",
            "c3bbbbbbbbbbbbbbbbbbbbbb",
          ],
          "operator": "u1",
        },
        Object {
          "cameras": Array [
            "c2bbbbbbbbbbbbbbbbbbbbbb",
          ],
          "operator": "u2",
        },
      ],
      "operators": Array [
        "u1",
        "u2",
      ],
      "shot": 1,
    },
    "undefinedb19999999999999999999999": null,
  },
}
`;

exports[`free unused presets after program Update works as planned: after reservation of four presets 2`] = `
Object {
  "currentPage": Object {},
  "dirtyCollections": Object {
    "presetAssistant": true,
    "projects": true,
  },
  "doingQuery": Object {
    "undefinedpresets{\\"camera\\":\\"c1bbbbbbbbbbbbbbbbbbbbbb\\",\\"$limit\\":15,\\"$skip\\":0,\\"$sort\\":{\\"number\\":1,\\"camera\\":1}}": false,
  },
  "lists": Object {
    "list.presetModel": Object {
      "data": Array [
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb098765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 0,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb198765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 1,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb298765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 2,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb398765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 3,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb498765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 4,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb598765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 5,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb698765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 6,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb798765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 7,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb898765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 8,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb998765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 9,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbb1098765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 10,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbb1198765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 11,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbb1298765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 12,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbb1398765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 13,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbb1498765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 14,
          "project": "-",
        },
      ],
      "limit": 15,
      "skip": 0,
      "total": 100,
    },
  },
  "makeQuery": [Function],
  "maxPage": Object {
    "list.presetModel": 7,
  },
  "pageSize": 15,
  "previousQuery": Object {
    "list.presetModel": Object {
      "$limit": 15,
      "$skip": 0,
      "$sort": Object {
        "camera": 1,
        "number": 1,
      },
      "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
    },
  },
}
`;

exports[`free unused presets after program Update works as planned: before anything 1`] = `
Object {
  "currentPage": Object {},
  "dirtyCollections": Object {},
  "doingQuery": Object {
    "undefinedpresets{\\"camera\\":\\"c1bbbbbbbbbbbbbbbbbbbbbb\\",\\"$limit\\":15,\\"$skip\\":0,\\"$sort\\":{\\"number\\":1,\\"camera\\":1}}": false,
  },
  "lists": Object {
    "list.presetModel": Object {
      "data": Array [
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb098765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 0,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb198765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 1,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb298765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 2,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb398765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 3,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb498765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 4,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb598765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 5,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb698765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 6,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb798765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 7,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb898765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 8,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbbb998765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 9,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbb1098765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 10,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbb1198765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 11,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbb1298765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 12,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbb1398765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 13,
          "project": "-",
        },
        Object {
          "_id": "c1bbbbbbbbbbbbbbb1498765",
          "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
          "number": 14,
          "project": "-",
        },
      ],
      "limit": 15,
      "skip": 0,
      "total": 100,
    },
  },
  "makeQuery": [Function],
  "maxPage": Object {
    "list.presetModel": 7,
  },
  "pageSize": 15,
  "previousQuery": Object {
    "list.presetModel": Object {
      "$limit": 15,
      "$skip": 0,
      "$sort": Object {
        "camera": 1,
        "number": 1,
      },
      "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
    },
  },
}
`;
